lib/socks.c

```diff
@@ -74,6 +74,7 @@ enum connect_t {
 struct socks_state {
   enum connect_t state;
   ssize_t outstanding;  /* send this many bytes more */
+  unsigned char buffer[8*1024]; /* more than enough */
```

Comment:
How did you pick the size?

Reply:
I have a crystal skull next to my dropping candles here in my lair.

As I read the code, one socks packets always seem to be smaller than 1k, but I might be wrong...therefore the assertions.

Reply:
I started replacing the fixes buffer with a bufq, but that quickly escalated all over the file and I was not sure if it would be worth it.

Reply:
I believe username, password and hostname are all limited to 256 bytes max, so 1K should probably be enough.